### PR TITLE
feat: allow passing validation set explicitly

### PR DESCRIPTION
**Pull request by JarbasAl**
i have been experimenting with model2vec and it is great! kudos for the awesome work

I wanted to use my own validation set but that was not exposed, this PR simply adds optional arguments to the `fit` method to allow passing your own validation data

### CODE REVIEW

File: model2vec/train/classifier.py [hunk 1]
```diff
@@ -166,11 +173,17 @@ def fit(
 
         self._initialize(y)
 
-        train_texts, validation_texts, train_labels, validation_labels = self._train_test_split(
-            X,
-            y,
-            test_size=test_size,
-        )
+        if X_val is not None and y_val is not None:
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply model2vec/train/classifier.py:176]: how do you want me to handle it when users pass only one of the vars? raise some specific Exception?

[the remark this answers is not in the document]
Comment (stephantul) [reply model2vec/train/classifier.py:176]: Yep, a `ValueError` is typically used to signify user error on input, so that should be fine.

**Comment by JarbasAl** [comment]
> This looks good, thanks for the PR. I can see that you didn't run pre-commit on this. Could you make the necessary changes and then run pre-commit before you commit?

sorry, i never used pre-commit before, i installed it and ran it before the last commit i pushed but it doesn't seem to have done anything?

**Comment by stephantul** [comment]
No worries, I can show you. From your output, I see you haven't staged the files, or have already pushed them. You can run `make install` from the root of the project, this will install and run pre-commit as a hook whenever you run `git commit`. This should make it block you from committing stuff that doesn't pass the pre-commit checks. Let me know if that works.

**Comment by JarbasAl** [comment]
i don't think my pycharm install is liking pre-commit... 

I can have a look later if that is a blocker for the PR and I'll redo it from scratch without pycharm in the mix 

The other requested changes have been pushed

**Comment by stephantul** [comment]
@JarbasAl , I added some more validation and tests, and ran the pre-commit. The tests are failing because of some unrelated transformers issue. Feel free to take a look if you like the changes.

**Review by stephantul — APPROVED** [review]
@JarbasAl Looks good, thanks for the contribution!

**Comment by stephantul** [comment]
The lates commits are to fix the failing CI
